packages/client/src/views/CorrectionForm/CorrectionReasonForm.tsx

```diff
@@ -0,0 +1,163 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import * as React from 'react'
+import { modifyApplication, IApplication } from '@client/applications'
+import { connect } from 'react-redux'
+import { WrappedComponentProps as IntlShapeProps, injectIntl } from 'react-intl'
+import { goBack } from '@client/navigation'
+import { IFormSection, IFormSectionData } from '@client/forms'
+import { replaceInitialValues } from '@client/views/RegisterForm/RegisterForm'
+import { ActionPageLight } from '@opencrvs/components/lib/interface'
+import styled from '@client/styledComponents'
+import { FormFieldGenerator } from '@client/components/form'
+import { PrimaryButton } from '@opencrvs/components/lib/buttons'
+import { buttonMessages } from '@client/i18n/messages'
+import { ErrorText } from '@opencrvs/components/lib/forms/ErrorText'
+import { getValidationErrorsForForm } from '@client/forms/validation'
+import { correctReasonSection } from '@client/forms/correction/reason'
+
+const FormSectionTitle = styled.h4`
+  ${({ theme }) => theme.fonts.h4Style};
+  color: ${({ theme }) => theme.colors.copy};
+  margin-top: 0px;
+  margin-bottom: 16px;
+`
+
+const ErrorWrapper = styled.div`
+  margin-top: -3px;
+  margin-bottom: 16px;
+`
+
+type IProps = {
+  application: IApplication
+}
+
+type IDispatchProps = {
+  goBack: typeof goBack
+  modifyApplication: typeof modifyApplication
+}
+
+type IFullProps = IProps & IDispatchProps & IntlShapeProps
+
+function getGroup(section: IFormSection, application: IApplication) {
+  const group = section.groups[0]
+
+  return {
+    ...group,
+    fields: replaceInitialValues(
+      group.fields,
+      application.data[section.id] || {},
+      application.data
+    )
+  }
+}
+
+function CorrectionReasonFormComponent(props: IFullProps) {
+  const [showError, setShowError] = React.useState(false)
+
+  const { application, intl, goBack } = props
+
+  const section = correctReasonSection
+
+  const group = getGroup(section, application)
+
+  const modifyApplication = (
+    sectionData: IFormSectionData,
+    activeSection: IFormSection,
+    application: IApplication
+  ) => {
+    props.modifyApplication({
+      ...application,
+      data: {
+        ...application.data,
+        [activeSection.id]: {
+          ...application.data[activeSection.id],
+          ...sectionData
+        }
+      }
+    })
+  }
+  const continueButtonHandler = () => {
+    const errors = getValidationErrorsForForm(
+      group.fields,
+      application.data[section.id] || {}
+    )
+
+    group.fields.forEach((field) => {
```

Comment:
You probably already know this, but if you'd do 

```
for(field of group.fields) {
```

instead of forEach, then the return on line 120 would break out of the loop completely, so now further fields wouldn't be handled